lib/ancestry/optimizers/pg.rb

```diff
@@ -0,0 +1,27 @@
+module Ancestry
+  module Optimizers
```

Comment:
I'd prefer to avoid the whole optimizer thing. This is simply how we do materialized path for postgres.
And this implementation is specific to postgres

just call it: `Ancestry::MaterializedPathPg`